static/js/compose_height.js

```diff
@@ -0,0 +1,54 @@
+// Making this similar to compose_state.js
+
+import $ from "jquery";
+
+import * as blueslip from "./blueslip";
+import * as compose_ui from "./compose_ui";
+
+const compose_textarea = $("#compose-textarea");
+let is_full_size = false; // true or false
+
+function get_expected_full_height() {
+    // This returns the full size of the compose box
+    // Panels can have different height and this calculates it accordignly
+    return window.innerHeight - 138 - Number.parseInt($("#panels").css("height"), 10);
+}
+
+// Some functions to handle the state explicitely
+export function set_is_full_size(is_full) {
+    is_full_size = is_full;
+}
+
+export function get_is_full_size() {
+    return is_full_size;
+}
+
+export function make_compose_box_full_size() {
+    compose_textarea.css("max-height", get_expected_full_height());
+    compose_textarea.css("height", get_expected_full_height());
+    $("#floating_recipient_bar").css("z-index", -1);
+}
```

Comment:
Would it work to change the `compose_textarea` CSS changes to instead just add a CSS class on the object that declares some properties on `compose_textarea.full_height` or similar (possibly with `!important` if needed?)

Reply:
Sure that would be a great idea.

Reply:
Well, I remember why I used this way. The panels are variable. Sometimes panels are there and sometimes they are not. So the height of the compose box really depends on this. So, I used javascript here as the height will be different.
Is there any way to do this with CSS?

Reply:
I have done this one with classes.